apps/root/src/pages/aggregator/swap-container/components/swap/index.tsx

```diff
@@ -1295,53 +1274,32 @@ const Swap = ({ isLoadingRoute, quotes, fetchOptions, swapOptionsError }: SwapPr
               isBuyOrder={isBuyOrder}
               isLoadingRoute={isLoadingRoute}
               transferTo={transferTo}
-              onOpenTransferTo={() => setShouldShowTransferModal(true)}
+              setShouldShowTransferModal={setShouldShowTransferModal}
               onShowSettings={onShowSettings}
               isApproved={isApproved}
               fromValue={fromValue}
               quotes={quotes}
               fetchOptions={fetchOptions}
               refreshQuotes={refreshQuotes}
               swapOptionsError={swapOptionsError}
+              swapButton={
+                <SwapButton
```

Comment:
this parent component receives most of the same props than the swap button, is there a reason why we are passing it through here and not just render it directly on the first step?